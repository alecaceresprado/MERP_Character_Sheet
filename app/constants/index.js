export * from './classes';
export * from './objects';
export * from './races';
