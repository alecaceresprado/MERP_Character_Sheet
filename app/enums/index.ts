export * from './races.enum';
export * from './objects.enum';
