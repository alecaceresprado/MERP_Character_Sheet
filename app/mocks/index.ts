export * from './characters.mock';
export * from './list.mock';
