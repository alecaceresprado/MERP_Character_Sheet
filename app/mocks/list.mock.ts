import { PlayerSummary } from 'models/player-summary.model';

export const playerList: PlayerSummary[] = [
  {
    id: '1',
    description: 'Braided long beard with charms with runes',
    avatarUrl:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
    name: 'Lothar Picco'
  },
  {
    id: '2',
    description: 'Braided long beard with charms with runes',
    avatarUrl:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
    name: 'Lothar Picco'
  },
  {
    id: '3',
    description: 'Braided long beard with charms with runes',
    avatarUrl:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
    name: 'Lothar Picco'
  },
  {
    id: '4',
    description: 'Braided long beard with charms with runes',
    avatarUrl:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
    name: 'Lothar Picco'
  }
];
