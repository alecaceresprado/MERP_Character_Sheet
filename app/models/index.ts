export * from './player.model';
export * from './repo.model';
export * from './player-summary.model';
