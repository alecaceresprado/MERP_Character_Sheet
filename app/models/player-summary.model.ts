export interface PlayerSummary {
  id: string;
  description: string;
  avatarUrl: string;
  name: string;
}
