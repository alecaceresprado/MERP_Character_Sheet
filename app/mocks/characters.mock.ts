import { RacesEnum, ObjectTypesEnum } from 'enums';
import { classes } from '../constants';
import { Player } from 'models/player.model';

export const characters: Player[] = [
  {
    id: 1,
    details: {
      name: 'Lothar Picco',
      nickName: 'Loti Picco',
      race: RacesEnum.NORSEMEN,
      height: '2 meters',
      weight: '110KG',
      hairColor: 'Red',
      eyesColor: 'Blue',
      physicalSpecial: 'Braided long beard with charms with runes',
      class: 'Viking',
      classId: classes.WARRIOR.classId,
      experience: 31751,
      level: 3
    },
    stats: {
      strength: {
        num: 102,
        normal: 35,
        race: 5,
        total: 40
      },
      agility: {
        num: 99,
        normal: 20,
        race: 0,
        total: 20
      },
      stamina: {
        num: 95,
        normal: 15,
        race: 5,
        total: 20
      },
      intellect: {
        num: 68,
        normal: 0,
        race: 0,
        total: 0
      },
      intuition: {
        num: 75,
        normal: 5,
        race: 0,
        total: 5
      },
      will: {
        num: 91,
        normal: 10,
        race: 0,
        total: 10
      },
      beauty: {
        num: 45,
        normal: 45,
        race: 0,
        total: 55
      }
    },
    skills: {
      armor: {
        noArmor: {
          skillPoints: 2
        },
        leather: {
          skillPoints: 1
        },
        hardenedLeather: {
          skillPoints: 0
        },
        chainMail: {
          skillPoints: 6
        },
        plate: {
          skillPoints: 1
        }
      },
      weapons: {
        edge: {
          skillPoints: 1
        },
        blunt: {
          skillPoints: 1
        },
        twoHands: {
          skillPoints: 0
        },
        throw: {
          skillPoints: 1
        },
        projectile: {
          skillPoints: 4
        },
        dualWeapons: {
          skillPoints: 8
        },
        pole: {
          skillPoints: 1
        }
      },
      exterior: {
        climb: {
          skillPoints: 2
        },
        mount: {
          skillPoints: 4
        },
        swim: {
          skillPoints: 4
        },
        tracking: {
          skillPoints: 1
        }
      },
      survival: {
        ambush: {
          skillPoints: 1
        },
        hide: {
          skillPoints: 4
        },
        lockPick: {
          skillPoints: 1
        },
        trapDismantle: {
          skillPoints: 1
        }
      },
      magic: {
        readRunes: {
          skillPoints: 1
        },
        useObject: {
          skillPoints: 1
        },
        aimedSpell: {
          skillPoints: 0
        }
      },
      global: {
        sight: {
          skillPoints: 6
        },
        health: {
          skillPoints: 13,
          pointValues: [5, 5, 6, 8, 8, 9, 7, 9, 9, 5, 3, 2, 2],
          sum: 78
        },
        mana: {
          skillPoints: 0,
          pointValues: [],
          sum: 0
        }
      },
      secondary: {
        blacksmith: {
          skillPoints: 7
        }
      }
    },
    inventory: [
      {
        name: 'regular chain mail',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor', 'chainMail']],
        modifier: 0
      },
      {
        name: 'enhanced arm guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['attack']],
        modifier: 0
      },
      {
        name: 'enhanced leg guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor']],
        modifier: 0
      },
      {
        name: 'throwing knives',
        quantity: 5,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'throw']],
        modifier: 0,
        description: 'personally crafted, shaped like raven feathers'
      },
      {
        name: 'winter vigil - magic axe',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [
          ['weapons', 'dualWeapons'],
          ['weapons', 'projectile']
        ],
        modifier: 10,
        description:
          'one handed axe: magic ability to eject axe`s head and then summon it back'
      },
      {
        name: 'regular sword',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'dualWeapons']],
        modifier: 0,
        description: 'one handed sword'
      },
      {
        name: 'special knives belt',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables to grab and throw on the same round'
      },
      {
        name: 'enhanced body cover (embozo)',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['survival', 'hide']],
        modifier: 10
      },
      {
        name: 'Charms neck lace',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'Raven charm'
      },
      {
        name: 'magic anklet (tobillera)',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables the user to cast the "world traveler" magic'
      },
      {
        name: 'odyn',
        quantity: 1,
        type: ObjectTypesEnum.PET,
        description: 'Raven, bound to me on my trials'
      },
      {
        name: 'gold coins',
        quantity: 1,
        type: ObjectTypesEnum.CURRENCY,
        description: ''
      }
    ]
  },
  {
    id: 2,
    details: {
      name: 'Legolas Greenleaf',
      nickName: 'Lembas',
      race: RacesEnum.SINDAR_ELF,
      height: '1,82 meters',
      weight: '25KG',
      hairColor: 'Blonde',
      eyesColor: 'Blue',
      physicalSpecial: 'Braided long beard with charms with runes',
      class: 'Viking',
      classId: classes.RANGER.classId,
      experience: 0,
      level: 1
    },
    stats: {
      strength: {
        num: 102,
        normal: 35,
        race: 5,
        total: 40
      },
      agility: {
        num: 99,
        normal: 20,
        race: 0,
        total: 20
      },
      stamina: {
        num: 95,
        normal: 15,
        race: 5,
        total: 20
      },
      intellect: {
        num: 68,
        normal: 0,
        race: 0,
        total: 0
      },
      intuition: {
        num: 75,
        normal: 5,
        race: 0,
        total: 5
      },
      will: {
        num: 91,
        normal: 10,
        race: 0,
        total: 10
      },
      beauty: {
        num: 45,
        normal: 45,
        race: 0,
        total: 55
      }
    },
    skills: {
      armor: {
        noArmor: {
          skillPoints: 2
        },
        leather: {
          skillPoints: 1
        },
        hardenedLeather: {
          skillPoints: 0
        },
        chainMail: {
          skillPoints: 6
        },
        plate: {
          skillPoints: 1
        }
      },
      weapons: {
        edge: {
          skillPoints: 1
        },
        blunt: {
          skillPoints: 1
        },
        twoHands: {
          skillPoints: 0
        },
        throw: {
          skillPoints: 1
        },
        projectile: {
          skillPoints: 4
        },
        dualWeapons: {
          skillPoints: 8
        },
        pole: {
          skillPoints: 1
        }
      },
      exterior: {
        climb: {
          skillPoints: 2
        },
        mount: {
          skillPoints: 4
        },
        swim: {
          skillPoints: 4
        },
        tracking: {
          skillPoints: 1
        }
      },
      survival: {
        ambush: {
          skillPoints: 1
        },
        hide: {
          skillPoints: 4
        },
        lockPick: {
          skillPoints: 1
        },
        trapDismantle: {
          skillPoints: 1
        }
      },
      magic: {
        readRunes: {
          skillPoints: 1
        },
        useObject: {
          skillPoints: 1
        },
        aimedSpell: {
          skillPoints: 0
        }
      },
      global: {
        sight: {
          skillPoints: 6
        },
        health: {
          skillPoints: 13,
          pointValues: [5, 5, 6, 8, 8, 9, 7, 9, 9, 5, 3, 2, 2],
          sum: 78
        },
        mana: {
          skillPoints: 0,
          pointValues: [],
          sum: 0
        }
      },
      secondary: {
        blacksmith: {
          skillPoints: 7
        }
      }
    },
    inventory: []
  },
  {
    id: 3,
    details: {
      name: 'Boromir son of Denethor',
      nickName: 'Boromir',
      race: RacesEnum.MAN,
      height: '1.93 meters',
      weight: '85KG',
      hairColor: 'Dark',
      eyesColor: 'Grey',
      physicalSpecial: 'Braided long beard with charms with runes',
      class: 'Viking',
      classId: classes.WARRIOR.classId,
      experience: 0,
      level: 0
    },
    stats: {
      strength: {
        num: 102,
        normal: 35,
        race: 5,
        total: 40
      },
      agility: {
        num: 99,
        normal: 20,
        race: 0,
        total: 20
      },
      stamina: {
        num: 95,
        normal: 15,
        race: 5,
        total: 20
      },
      intellect: {
        num: 68,
        normal: 0,
        race: 0,
        total: 0
      },
      intuition: {
        num: 75,
        normal: 5,
        race: 0,
        total: 5
      },
      will: {
        num: 91,
        normal: 10,
        race: 0,
        total: 10
      },
      beauty: {
        num: 45,
        normal: 45,
        race: 0,
        total: 55
      }
    },
    skills: {
      armor: {
        noArmor: {
          skillPoints: 2
        },
        leather: {
          skillPoints: 1
        },
        hardenedLeather: {
          skillPoints: 0
        },
        chainMail: {
          skillPoints: 6
        },
        plate: {
          skillPoints: 1
        }
      },
      weapons: {
        edge: {
          skillPoints: 1
        },
        blunt: {
          skillPoints: 1
        },
        twoHands: {
          skillPoints: 0
        },
        throw: {
          skillPoints: 1
        },
        projectile: {
          skillPoints: 4
        },
        dualWeapons: {
          skillPoints: 8
        },
        pole: {
          skillPoints: 1
        }
      },
      exterior: {
        climb: {
          skillPoints: 2
        },
        mount: {
          skillPoints: 4
        },
        swim: {
          skillPoints: 4
        },
        tracking: {
          skillPoints: 1
        }
      },
      survival: {
        ambush: {
          skillPoints: 1
        },
        hide: {
          skillPoints: 4
        },
        lockPick: {
          skillPoints: 1
        },
        trapDismantle: {
          skillPoints: 1
        }
      },
      magic: {
        readRunes: {
          skillPoints: 1
        },
        useObject: {
          skillPoints: 1
        },
        aimedSpell: {
          skillPoints: 0
        }
      },
      global: {
        sight: {
          skillPoints: 6
        },
        health: {
          skillPoints: 13,
          pointValues: [5, 5, 6, 8, 8, 9, 7, 9, 9, 5, 3, 2, 2],
          sum: 78
        },
        mana: {
          skillPoints: 0,
          pointValues: [],
          sum: 0
        }
      },
      secondary: {
        blacksmith: {
          skillPoints: 7
        }
      }
    },
    inventory: [
      {
        name: 'regular chain mail',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor', 'chainMail']],
        modifier: 0
      },
      {
        name: 'enhanced arm guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['attack']],
        modifier: 0
      },
      {
        name: 'enhanced leg guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor']],
        modifier: 0
      },
      {
        name: 'throwing knives',
        quantity: 5,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'throw']],
        modifier: 0,
        description: 'personally crafted, shaped like raven feathers'
      },
      {
        name: 'winter vigil - magic axe',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [
          ['weapons', 'dualWeapons'],
          ['weapons', 'projectile']
        ],
        modifier: 10,
        description:
          'one handed axe: magic ability to eject axe`s head and then summon it back'
      },
      {
        name: 'regular sword',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'dualWeapons']],
        modifier: 0,
        description: 'one handed sword'
      },
      {
        name: 'special knives belt',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables to grab and throw on the same round'
      },
      {
        name: 'enhanced body cover (embozo)',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['survival', 'hide']],
        modifier: 10
      },
      {
        name: 'Charms neck lace',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'Raven charm'
      },
      {
        name: 'magic anklet (tobillera)',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables the user to cast the "world traveler" magic'
      },
      {
        name: 'odyn',
        quantity: 1,
        type: ObjectTypesEnum.PET,
        description: 'Raven, bound to me on my trials'
      },
      {
        name: 'gold coins',
        quantity: 1,
        type: ObjectTypesEnum.CURRENCY,
        description: ''
      }
    ]
  },
  {
    id: 4,
    details: {
      name: 'Aragorn II',
      nickName: 'Estel',
      race: RacesEnum.DUNEDAIN,
      height: '1.85 meters',
      weight: '80KG',
      hairColor: 'Dark',
      eyesColor: 'Grey',
      physicalSpecial: 'Braided long beard with charms with runes',
      class: 'Viking',
      classId: classes.RANGER.classId,
      experience: 0,
      level: 0
    },
    stats: {
      strength: {
        num: 102,
        normal: 35,
        race: 5,
        total: 40
      },
      agility: {
        num: 99,
        normal: 20,
        race: 0,
        total: 20
      },
      stamina: {
        num: 95,
        normal: 15,
        race: 5,
        total: 20
      },
      intellect: {
        num: 68,
        normal: 0,
        race: 0,
        total: 0
      },
      intuition: {
        num: 75,
        normal: 5,
        race: 0,
        total: 5
      },
      will: {
        num: 91,
        normal: 10,
        race: 0,
        total: 10
      },
      beauty: {
        num: 45,
        normal: 45,
        race: 0,
        total: 55
      }
    },
    skills: {
      armor: {
        noArmor: {
          skillPoints: 2
        },
        leather: {
          skillPoints: 1
        },
        hardenedLeather: {
          skillPoints: 0
        },
        chainMail: {
          skillPoints: 6
        },
        plate: {
          skillPoints: 1
        }
      },
      weapons: {
        edge: {
          skillPoints: 1
        },
        blunt: {
          skillPoints: 1
        },
        twoHands: {
          skillPoints: 0
        },
        throw: {
          skillPoints: 1
        },
        projectile: {
          skillPoints: 4
        },
        dualWeapons: {
          skillPoints: 8
        },
        pole: {
          skillPoints: 1
        }
      },
      exterior: {
        climb: {
          skillPoints: 2
        },
        mount: {
          skillPoints: 4
        },
        swim: {
          skillPoints: 4
        },
        tracking: {
          skillPoints: 1
        }
      },
      survival: {
        ambush: {
          skillPoints: 1
        },
        hide: {
          skillPoints: 4
        },
        lockPick: {
          skillPoints: 1
        },
        trapDismantle: {
          skillPoints: 1
        }
      },
      magic: {
        readRunes: {
          skillPoints: 1
        },
        useObject: {
          skillPoints: 1
        },
        aimedSpell: {
          skillPoints: 0
        }
      },
      global: {
        sight: {
          skillPoints: 6
        },
        health: {
          skillPoints: 13,
          pointValues: [5, 5, 6, 8, 8, 9, 7, 9, 9, 5, 3, 2, 2],
          sum: 78
        },
        mana: {
          skillPoints: 0,
          pointValues: [],
          sum: 0
        }
      },
      secondary: {
        blacksmith: {
          skillPoints: 7
        }
      }
    },
    inventory: [
      {
        name: 'regular chain mail',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor', 'chainMail']],
        modifier: 0
      },
      {
        name: 'enhanced arm guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['attack']],
        modifier: 0
      },
      {
        name: 'enhanced leg guards',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['armor']],
        modifier: 0
      },
      {
        name: 'throwing knives',
        quantity: 5,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'throw']],
        modifier: 0,
        description: 'personally crafted, shaped like raven feathers'
      },
      {
        name: 'winter vigil - magic axe',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [
          ['weapons', 'dualWeapons'],
          ['weapons', 'projectile']
        ],
        modifier: 10,
        description:
          'one handed axe: magic ability to eject axe`s head and then summon it back'
      },
      {
        name: 'regular sword',
        quantity: 1,
        type: ObjectTypesEnum.WEAPON,
        skill: [['weapons', 'dualWeapons']],
        modifier: 0,
        description: 'one handed sword'
      },
      {
        name: 'special knives belt',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables to grab and throw on the same round'
      },
      {
        name: 'enhanced body cover (embozo)',
        type: ObjectTypesEnum.SKILL_ENHANCER,
        skill: [['survival', 'hide']],
        modifier: 10
      },
      {
        name: 'Charms neck lace',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'Raven charm'
      },
      {
        name: 'magic anklet (tobillera)',
        quantity: 1,
        type: ObjectTypesEnum.CUSTOM,
        description: 'enables the user to cast the "world traveler" magic'
      },
      {
        name: 'odyn',
        quantity: 1,
        type: ObjectTypesEnum.PET,
        description: 'Raven, bound to me on my trials'
      },
      {
        name: 'gold coins',
        quantity: 1,
        type: ObjectTypesEnum.CURRENCY,
        description: ''
      }
    ]
  }
];
