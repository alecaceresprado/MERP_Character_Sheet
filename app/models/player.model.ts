import { RacesEnum } from '../enums'

export interface Player {
    details: PlayerDetails;
    stats: PlayerStats;
    skills: PlayerSkills;
    inventory: PlayerInventory[];
}

export interface PlayerDetails {
    name: string,
    nickName: string,
    race: RacesEnum.NORSEMEN,
    height: string,
    weight: string,
    hairColor: string,
    eyesColor: string,
    physicalSpecial: string,
    class: string,
    classId: string,
    experience: 31751,
    level: 3,
}

export interface PlayerStats {
    strength: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    agility: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    stamina: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    intellect: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    intuition: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    will: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    beauty: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
}

export interface PlayerSkills {
    armor: {
        noArmor: {
            skillPoints: number,
        },
        leather: {
            skillPoints: number,
        },
        hardenedLeather: {
            skillPoints: number,
        },
        chainMail: {
            skillPoints: number,
        },
        plate: {
            skillPoints: number,
        },
    },
    weapons: {
        edge: {
            skillPoints: number,
        },
        blunt: {
            skillPoints: number,
        },
        twoHands: {
            skillPoints: number,
        },
        throw: {
            skillPoints: number,
        },
        projectile: {
            skillPoints: number,
        },
        dualWeapons: {
            skillPoints: number,
        },
        pole: {
            skillPoints: number,
        },
    },
    exterior: {
        climb: {
            skillPoints: number,
        },
        mount: {
            skillPoints: number,
        },
        swim: {
            skillPoints: number,
        },
        tracking: {
            skillPoints: number,
        },
    },
    survival: {
        ambush: {
            skillPoints: number,
        },
        hide: {
            skillPoints: number,
        },
        lockPick: {
            skillPoints: number,
        },
        trapDismantle: {
            skillPoints: number,
        },
    },
    magic: {
        readRunes: {
            skillPoints: number,
        },
        useObject: {
            skillPoints: number,
        },
        aimedSpell: {
            skillPoints: number,
        },
    },
    global: {
        sight: {
            skillPoints: number,
        },
        health: {
            skillPoints: number,
            pointValues: number[],
            sum: number,
        },
        mana: {
            skillPoints: number,
            pointValues: number[],
            sum: number,
        },
    },
    secondary: {
        blacksmith: {
            skillPoints: number,
        },
    },
}

export interface PlayerInventory {
    name: string;
    type: string;
    skill?: string[][];
    modifier?: number;
    description?: string;
    quantity?: number;
}
