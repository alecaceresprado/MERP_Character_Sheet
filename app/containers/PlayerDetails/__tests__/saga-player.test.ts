/**
 * Tests for HomePage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { AppActions, AppActionTypes } from 'state/app/app.actions';
import getPlayerDetailData, { getPlayersDetail } from '../saga-player';
import { PlayerSummary, Player } from 'models';
import { characters } from 'mocks';

describe('getPlayersDetail Saga', () => {
    let getPlayerDetailGenerator;

    // We have to test twice, once for a successful load and once for an unsuccessful one
    // so we do all the stuff that happens beforehand automatically in the beforeEach
    beforeEach(() => {
        getPlayerDetailGenerator = getPlayersDetail(AppActions.loadPlayerDetail('1'));

        const selectDescriptor = getPlayerDetailGenerator.next().value;
        expect(selectDescriptor).toMatchSnapshot();
    });

    it('should dispatch the playerDetailsLoaded action if it requests the data successfully', () => {
        const response = characters[0];
        const putDescriptor = getPlayerDetailGenerator.next(response).value;
        expect(putDescriptor).toEqual(
            put(AppActions.playerDetailsLoaded(response))
        );
    });

    it('should call the playerDetailsLoadingError action if the response errors', () => {
        const response = new Error('Some error');
        const putDescriptor = getPlayerDetailGenerator.throw(response).value;
        expect(putDescriptor).toEqual(
            put(AppActions.playerDetailsLoadingError(response))
        );
    });
});

describe('getPlayerDetailData Saga', () => {
    const getPlayerDetailDataSaga = getPlayerDetailData();

    it('should start task to watch for LOAD_PLAYER_DETAIL action', () => {
        const takeLatestDescriptor = getPlayerDetailDataSaga.next().value;
        expect(takeLatestDescriptor).toEqual(
            takeLatest(AppActionTypes.LOAD_PLAYER_DETAIL, getPlayersDetail)
        );
    });
});
