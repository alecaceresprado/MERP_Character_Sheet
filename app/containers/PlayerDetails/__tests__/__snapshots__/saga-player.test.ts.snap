// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPlayersDetail Saga should call the playerDetailsLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/mockPlayer/1",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getPlayersDetail Saga should dispatch the playerDetailsLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/mockPlayer/1",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
